strongbox-aql/src/main/antlr4/AQL.g4

```diff
@@ -0,0 +1,241 @@
+grammar AQL;
+
+@header {
+    package org.carlspring.strongbox.aql.grammar;
+}
+
+query
+:
+    queryExp+ orderExp? pageExp? EOF
+;
+
+pageExp
+:
+    PAGE_SKIP COLON NUMBER
+;
+
+orderExp
+:
+    orderDirection COLON orderValue
+;
+
+orderValue
+:
+    STORAGE
+    | REPOSITORY
+    | LAYOUT
+    | VERSION
+    | TAG
+    | AGE
+    | IDENTIFIER
+;
+
+orderDirection
+:
+    ASC
+    | DESC
+;
+
+queryExp
+:
+    tokenPrefix? ROUND_BRACKET_LEFT vNesteedQueryExp = queryExp
+    ROUND_BRACKET_RIGHT
+    | vQueryExpLeft = queryExp logicalOp? vQueryExpRight = queryExp
+    | tokenPrefix? tokenExp
+;
+
+tokenExp
+:
+    tokenKey COLON tokenValue
+;
+
+tokenPrefix
+:
+    PLUS
+    | NEGATION
+    | PLUS NEGATION
+;
+
+tokenKey
+:
+    tokenKeyword
+    | IDENTIFIER
+;
+
+tokenValue
+:
+    IDENTIFIER
+    | VALUE
+    | STRING
+;
+
+tokenKeyword
+:
+    STORAGE
+    | REPOSITORY
+    | LAYOUT
+    | VERSION
+    | TAG
+    | FROM
+    | TO
+    | AGE
+;
+
+logicalOp
+:
+    and
+    | or
+;
+
+and
+:
+    AND
+    | AMP
+    | DOUBLE_AMP
+;
+
+or
+:
+    OR
+    | PIPE
+    | DOUBLE_PIPE
+;
+
+STORAGE
+:
+    'storage'
```

Comment:
Correct me, if I am wrong (about this keyword and the ones below), but does it mean that it doesn't support uppercase? If so, could you please add it as well (here and wherever else required)? Thanks!

Reply:
I suggest to make case insensitive grammar with upcoming tasks.

Reply:
Could you please raise a new issue for this on Github with a proper explanation?

Reply:
Could you please raise a new issue for this on Github with a proper explanation?